flink-sql-runner/src/main/java/com/github/streamshub/flink/SqlRunner.java

```diff
@@ -0,0 +1,142 @@
+        if (args.length != 1) {
+            throw new Exception("Exactly 1 argument is expected.");
+        }
+
+        var statements = parseStatements(args[0]);
+
+        EnvironmentSettings settings = EnvironmentSettings
+                .newInstance()
+                .inStreamingMode()
+                .build();
+        var tableEnv = TableEnvironment.create(settings);
+        LOG.debug("TableEnvironment config: " + tableEnv.getConfig().toMap());
+
+        for (String statement : statements) {
+            var processedStatement = interpolateSecrets(statement);
+            Matcher setMatcher = SET_STATEMENT_PATTERN.matcher(statement.trim());
+
+            if (setMatcher.matches()) {
+                // Handle SET statements
+                String key = setMatcher.group(1);
+                String value = setMatcher.group(2);
+                LOG.debug("Setting configurations:\n{}={}", key, value);
+                tableEnv.getConfig().getConfiguration().setString(key, value);
+            } else {
+                LOG.info("Executing:\n{}", statement);
+                tableEnv.executeSql(processedStatement);
+            }
+        }
+    }
+
+    private static List<String> parseStatements(String rawStatements) {
+        var formatted = formatSqlStatements(rawStatements.trim());
+
+        var statements = new ArrayList<String>();
+        StringBuilder current = new StringBuilder();
+        Matcher matcher = STATEMENT_SET_PATTERN.matcher(formatted);
+
+        String statementSet = "";
+        String otherStatements = formatted;
+
+        if (matcher.find()) {
+            statementSet = matcher.group(1);
+            otherStatements = formatted.replace(statementSet, "").trim();
+        }
+
+        for (char c : otherStatements.toCharArray()) {
+            if (c == STATEMENT_DELIMITER.charAt(0)) {
+                current.append(c);
+                statements.add(current.toString().trim());
+                current = new StringBuilder();
+            } else {
+                current.append(c);
+            }
+        }
+
+        if (statementSet.length() > 0) {
+            statements.add(statementSet);
+        }
+
+        return statements;
+    }
+
+    private static String interpolateSecrets(String statement) {
+        return KubernetesSecretReplacer.replaceSecrets(statement)
+                .replaceAll(BEGIN_CERTIFICATE, ESCAPED_BEGIN_CERTIFICATE)
+                .replaceAll(END_CERTIFICATE, ESCAPED_END_CERTIFICATE)
+                .replaceAll(BEGIN_PRIVATE_KEY, ESCAPED_BEGIN_PRIVATE_KEY)
+                .replaceAll(END_PRIVATE_KEY, ESCAPED_END_PRIVATE_KEY)
+                .replaceAll(COMMENT_PATTERN, "")
+                .replaceAll(ESCAPED_BEGIN_CERTIFICATE, BEGIN_CERTIFICATE)
+                .replaceAll(ESCAPED_END_CERTIFICATE, END_CERTIFICATE)
+                .replaceAll(ESCAPED_BEGIN_PRIVATE_KEY, BEGIN_PRIVATE_KEY)
+                .replaceAll(ESCAPED_END_PRIVATE_KEY, END_PRIVATE_KEY);
```

Comment:
I'm not sure I see a distinction between what the secret replacer returns and this replacement phase. I suspect there is a wrapper or at least a further separation between the runner and the statement sanitiser.

Reply:
You are right, I think this part might not be actually necessary. The original SqlRunner example sanitises the statements from the script so that when removing comments, the cert prefix and suffix don't get mixed up and removed. As we are passing the statements as an argument (not a script file), I don't think we expect any comments to be included. I think I will remove these replacements.